File: Niveau 3/Niveau 3.11/opposés.hs
--------------------------------------------------------------------------------
-- Name:        Nombres opposés
-- Purpose:     France-IOI niveau 3.11
--
-- Author:      Sébastien
--
-- Created:     29/10/2022
--------------------------------------------------------------------------------

import Data.Containers.ListUtils
import Data.List

main :: IO ()
main = do
    _ <- readLn :: IO Int
    nombres <- fmap (map read . words) getLine :: IO [Int]
    let positifs = sort [ n | n <- nombres, n > 0]
        negatifs = sort [-n | n <- nombres, n < 0]
    -- print $ length (nubInt (intersect positifs negatifs))
    print $ nbCommuns 0 positifs negatifs
    
    
nbCommuns :: Int -> [Int] -> [Int] -> Int
nbCommuns _ [] _ = 0
nbCommuns _ _ [] = 0
nbCommuns précédent (x:xs) (y:ys)
    | x < y = nbCommuns précédent xs (y:ys)
    | x > y = nbCommuns précédent (x:xs) ys
    | x == y && x == précédent = nbCommuns précédent xs ys
    | otherwise = 1 + nbCommuns x xs ys

File: Niveau 3/Niveau 3.11/opposés.py
#-------------------------------------------------------------------------------
# Name:        Nombres opposés
# Purpose:     France-IOI niveau 3.11
#
# Author:      Sébastien
#
# Created:     29/10/2022
#-------------------------------------------------------------------------------

def main():
    _ = int(input())
    nombres = list(map(int, input().split()))
    positifs = sorted([ n for n in nombres if n > 0])
    negatifs = sorted([-n for n in nombres if n < 0])

    n = 0
    i = 0
    j = 0
    valeur_precedente = 0
    while i < len(positifs) and j < len(negatifs):
        if positifs[i] == negatifs[j]:
            if positifs[i] != valeur_precedente:
                n += 1
                valeur_precedente = positifs[i]
            i += 1
            j += 1
        elif positifs[i] < negatifs[j]:
            i += 1
        else:
            j += 1

    print(n)

if __name__ == '__main__':
    main()

"""
15
-3 4 2 8 9 1 -3 -8 -4 2 8 2 -8 1 3
"""